docs/explanations/session-overlay-state-machine.md

```
@@ -0,0 +1,145 @@
+
+### Minimized Variant Changes
+**Trigger**: User interaction within `MinimizedSessionBar`
+
+```
+handleSetMinimizedVariant(variant) in App.tsx
+  → window.api.setMinimizedOverlayVariant(variant)
+  → [IPC] overlayHandlers.ts: setMinimizedVariant(variant), switchOverlayMode('minimized')
+  → [Window resizes: stays minimized but changes dimensions]
+  → setOverlayMode('minimized'), setMinimizedVariant(variant)
+```
+
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)